lib/article_json/export/amp/elements/embed.rb

```diff
@@ -20,20 +20,40 @@ def embed_node
             end
           end
 
-          def amp_tag
-            {
-              vimeo_video: 'amp-vimeo',
-              youtube_video: 'amp-youtube',
-            }[@element.embed_type.to_sym]
+          def type_specific_node
+            case @element.embed_type
+            when :youtube_video
+              youtube_node
+            when :vimeo_video
+              vimeo_node
+            when :facebook_video
+              facebook_node
+            end
           end
 
-          def type_specific_node
-            create_element(amp_tag,
+          def youtube_node
+            create_element('amp-youtube',
+                           'data-videoid' => @element.embed_id,
```

Comment:
Since the other keys are symbols here you can also use `'my-key': <value>` instead of `=>` and then `'data-videoid'` will also be a symbol-key :)

```ruby
create_element('amp-youtube',
               'data-videoid': @element.embed_id,
               width: default_width,
               height: default_height)
```

Reply:
Oh, that's nice, I'll change it in next PR, thanks!

Reply:
@nicolas-fricke Changed here https://github.com/Devex/article_json/pull/85

Reply:
I tried that, but code climate complains, because it doesn't seem compatible with ruby 2.1
https://codeclimate.com/github/Devex/article_json/pull/85

So I'm reverting it.

Reply:
Oh, I didn't know that the Gem was 2.1 compatible.. I thought that we'd only target >2.3 (since I thought I also seen usages of the safe navigation operator `foo&.bar` which is >2.3 only).

Reply:
You are right, Daniel told me yesterday we are using ruby >= 2.3, but I got confused by the code climate error message.